### DISCUSSION TITLE
Galery remains open when use browser navigation

### DISCUSSION THREAD

**Discussion by ugurcanerkal** [dimsemenov/PhotoSwipe#2117] [Q&A]
Hi guys, I just wanted to ask if this is a known issue, when I click on an image and open the gallery, if I use browser navigation, the gallery is still open, I can still zoom, etc. Is that a known issue?

**Comment by ugurcanerkal** [comment]
> Hi guys, I just wanted to ask if this is a known issue, when I click on an image and open the gallery, if I use browser navigation, the gallery is still open, I can still zoom, etc. Is that a known issue?

I just tested in the PhotoSwipe site and it is also working like that. So I assume this is a known issue?

**Comment by dimsemenov** [comment]
Hi, I'm not sure what are you referring to by the "browser navigation". Can you please elaborate, or provide some screenshot of an issue.

**Reply by MentalGear** [reply]
I think he means Client-side navigation, as in a SPA. I noticed the same behaviour in sveltekit.
E.g. open a lightbox, navigate back, the lightbox stays instead of closing.

Used this to fix it:
```js
    window.addEventListener('popstate', function (event) {
        // close on browser navigation
        if (lightbox.pswp) {
            lightbox.pswp.options.showHideAnimationType = 'fade'
            lightbox.pswp.close()
        }
    })
```

Could maybe also be added to the core?

**Reply by acwolff** [reply]
@MentalGear please explain what you mean by

in a **SPA**. I noticed the same behaviour in **sveltekit**.

**Reply by MentalGear** [reply]
Single page application.

**Reply by acwolff** [reply]
What is a Single Page application? Please give a link to such a page.
And what is sveltekit?
